Authentication/AuthenticationBootstrapper.swift

```diff
@@ -12,8 +12,9 @@ import Foundation
     Bootstrapper to start the application.
     Takes care of starting the authentication process before the main View Controller of the app when necessary,
     and after the user logs out.
+    The authentication process includes login, signup and recover password logic.
 */
-public class AuthenticationBootstraper<User: UserType, SessionService: SessionServiceType where SessionService.User == User> {
+public class AuthenticationBootstrapper<User: UserType, SessionService: SessionServiceType where SessionService.User == User> {
```

Comment:
We can have a createLoginViewDelegate to continue with the pattern, and setting it to the view before return it.
I don't think there's any "nice" way, we will end up having so many create methods or public vars. So any of the two are fine with me.